{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotating the variants of genes of the patient from the gencode database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_annotations_filename = \"gencode.v18.annotation.gtf\"\n",
    "variant_filename = 'Metabolic_variants.vcf'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Extracted GTF attributes: ['gene_id', 'transcript_id', 'gene_type', 'gene_status', 'gene_name', 'transcript_type', 'transcript_status', 'transcript_name', 'level', 'havana_gene', 'tag', 'havana_transcript', 'exon_number', 'exon_id', 'ont', 'ccdsid']\n"
     ]
    }
   ],
   "source": [
    "from gtfparse import read_gtf\n",
    "# returns GTF with essential columns such as \"feature\", \"seqname\", \"start\", \"end\"\n",
    "# alongside the names of any optional keys which appeared in the attribute column\n",
    "df = read_gtf(gene_annotations_filename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>source</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>frame</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>...</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>level</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>tag</th>\n",
       "      <th>havana_transcript</th>\n",
       "      <th>exon_number</th>\n",
       "      <th>exon_id</th>\n",
       "      <th>ont</th>\n",
       "      <th>ccdsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14412</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>2</td>\n",
       "      <td>OTTHUMG00000000961.2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>14363</td>\n",
       "      <td>29806</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>2</td>\n",
       "      <td>OTTHUMG00000000958.1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>...</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>2</td>\n",
       "      <td>OTTHUMG00000000959.2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>2</td>\n",
       "      <td>OTTHUMG00000000960.1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>52473</td>\n",
       "      <td>54936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>2</td>\n",
       "      <td>OTTHUMG00000185779.1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    seqname  source feature  start    end  score strand frame  \\\n",
       "0      chr1  HAVANA    gene  11869  14412    NaN      +     0   \n",
       "21     chr1  HAVANA    gene  14363  29806    NaN      -     0   \n",
       "82     chr1  HAVANA    gene  29554  31109    NaN      +     0   \n",
       "92     chr1  HAVANA    gene  34554  36081    NaN      -     0   \n",
       "100    chr1  HAVANA    gene  52473  54936    NaN      +     0   \n",
       "\n",
       "               gene_id      transcript_id  ...   transcript_status  \\\n",
       "0    ENSG00000223972.4  ENSG00000223972.4  ...               KNOWN   \n",
       "21   ENSG00000227232.4  ENSG00000227232.4  ...               KNOWN   \n",
       "82   ENSG00000243485.2  ENSG00000243485.2  ...               NOVEL   \n",
       "92   ENSG00000237613.2  ENSG00000237613.2  ...               KNOWN   \n",
       "100  ENSG00000268020.2  ENSG00000268020.2  ...               KNOWN   \n",
       "\n",
       "    transcript_name level           havana_gene tag havana_transcript  \\\n",
       "0           DDX11L1     2  OTTHUMG00000000961.2                         \n",
       "21           WASH7P     2  OTTHUMG00000000958.1                         \n",
       "82       MIR1302-11     2  OTTHUMG00000000959.2                         \n",
       "92          FAM138A     2  OTTHUMG00000000960.1                         \n",
       "100          OR4G4P     2  OTTHUMG00000185779.1                         \n",
       "\n",
       "    exon_number exon_id ont ccdsid  \n",
       "0                                   \n",
       "21                                  \n",
       "82                                  \n",
       "92                                  \n",
       "100                                 \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filter DataFrame to gene entries on chrY\n",
    "df_genes = df[df[\"feature\"] == \"gene\"]\n",
    "df_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 57445 entries, 0 to 2614559\n",
      "Data columns (total 24 columns):\n",
      "seqname              57445 non-null object\n",
      "source               57445 non-null object\n",
      "feature              57445 non-null object\n",
      "start                57445 non-null int64\n",
      "end                  57445 non-null int64\n",
      "score                0 non-null float32\n",
      "strand               57445 non-null object\n",
      "frame                57445 non-null object\n",
      "gene_id              57445 non-null object\n",
      "transcript_id        57445 non-null object\n",
      "gene_type            57445 non-null object\n",
      "gene_status          57445 non-null object\n",
      "gene_name            57445 non-null object\n",
      "transcript_type      57445 non-null object\n",
      "transcript_status    57445 non-null object\n",
      "transcript_name      57445 non-null object\n",
      "level                57445 non-null object\n",
      "havana_gene          57445 non-null object\n",
      "tag                  57445 non-null object\n",
      "havana_transcript    57445 non-null object\n",
      "exon_number          57445 non-null object\n",
      "exon_id              57445 non-null object\n",
      "ont                  57445 non-null object\n",
      "ccdsid               57445 non-null object\n",
      "dtypes: float32(1), int64(2), object(21)\n",
      "memory usage: 10.7+ MB\n"
     ]
    }
   ],
   "source": [
    "df_genes.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand',\n",
       "       'frame', 'gene_id', 'transcript_id', 'gene_type', 'gene_status',\n",
       "       'gene_name', 'transcript_type', 'transcript_status', 'transcript_name',\n",
       "       'level', 'havana_gene', 'tag', 'havana_transcript', 'exon_number',\n",
       "       'exon_id', 'ont', 'ccdsid'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_genes.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['chr1', 'chr2', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8',\n",
       "       'chr9', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15',\n",
       "       'chr16', 'chr17', 'chr18', 'chr19', 'chr20', 'chr21', 'chr22',\n",
       "       'chrX', 'chrY', 'chrM'], dtype=object)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_genes['seqname'].unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Look into the list of variants and find correspondance to the gene from the gencode file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import vcf\n",
    "import pandas as pd\n",
    "\n",
    "vcf_reader = vcf.Reader(open(variant_filename, 'r'))\n",
    "variants = pd.DataFrame(columns=['CHROM', 'start', 'end', 'strand', 'gene_id', 'transcript_id', 'gene_type', 'gene_status',\n",
    "       'gene_name', 'transcript_type', 'transcript_status', 'transcript_name', 'POS','REF','ALT','FILTER'])\n",
    "\n",
    "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "counter = 0\n",
    "for record in vcf_reader:\n",
    "    chrom = record.CHROM\n",
    "    pos = record.POS\n",
    "    ref = record.REF\n",
    "    alt = record.ALT\n",
    "    filter_pass = record.FILTER\n",
    "    for row in df_genes.itertuples():\n",
    "        #print(str(chrom), row.seqname[3:])\n",
    "        there_is_entry = str(chrom) == row.seqname[3:] and pos > row.start and pos < row.end\n",
    "        if there_is_entry:\n",
    "            new_row = {'POS': pos, 'start': row.start, 'end': row.end, 'strand': row.strand, 'gene_id' : row.gene_id, 'transcript_id' : row.transcript_id, 'gene_type' : row.gene_type, 'gene_status': row.gene_status, 'gene_name' : row.gene_name, 'transcript_type' : row.transcript_type, 'transcript_status': row.transcript_status, 'transcript_name' : row.transcript_name, 'CHROM' : chrom, 'POS': pos,'REF':ref,'ALT':alt,'FILTER':filter_pass}\n",
    "            variants = variants.append(new_row, ignore_index=True)\n",
    "            counter += 1\n",
    "            #print(counter)\n",
    "            if counter % 100 == 0:\n",
    "                print(variants.tail())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_disease = pd.read_table('curated_gene_disease_associations.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>DSI</th>\n",
       "      <th>DPI</th>\n",
       "      <th>diseaseId</th>\n",
       "      <th>diseaseName</th>\n",
       "      <th>diseaseType</th>\n",
       "      <th>diseaseClass</th>\n",
       "      <th>diseaseSemanticType</th>\n",
       "      <th>score</th>\n",
       "      <th>EI</th>\n",
       "      <th>YearInitial</th>\n",
       "      <th>YearFinal</th>\n",
       "      <th>NofPmids</th>\n",
       "      <th>NofSnps</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>81741</th>\n",
       "      <td>109580095</td>\n",
       "      <td>HBB-LCR</td>\n",
       "      <td>0.815</td>\n",
       "      <td>0.069</td>\n",
       "      <td>C0002875</td>\n",
       "      <td>Cooley's anemia</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81742</th>\n",
       "      <td>109580095</td>\n",
       "      <td>HBB-LCR</td>\n",
       "      <td>0.815</td>\n",
       "      <td>0.069</td>\n",
       "      <td>C0005283</td>\n",
       "      <td>beta Thalassemia</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81743</th>\n",
       "      <td>109580095</td>\n",
       "      <td>HBB-LCR</td>\n",
       "      <td>0.815</td>\n",
       "      <td>0.069</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81744</th>\n",
       "      <td>109580095</td>\n",
       "      <td>HBB-LCR</td>\n",
       "      <td>0.815</td>\n",
       "      <td>0.069</td>\n",
       "      <td>C0085578</td>\n",
       "      <td>Thalassemia Minor</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81745</th>\n",
       "      <td>109580095</td>\n",
       "      <td>HBB-LCR</td>\n",
       "      <td>0.815</td>\n",
       "      <td>0.069</td>\n",
       "      <td>C0271979</td>\n",
       "      <td>Thalassemia Intermedia</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          geneId geneSymbol    DSI    DPI diseaseId             diseaseName  \\\n",
       "81741  109580095    HBB-LCR  0.815  0.069  C0002875         Cooley's anemia   \n",
       "81742  109580095    HBB-LCR  0.815  0.069  C0005283        beta Thalassemia   \n",
       "81743  109580095    HBB-LCR  0.815  0.069  C0019025    Hemoglobin F Disease   \n",
       "81744  109580095    HBB-LCR  0.815  0.069  C0085578       Thalassemia Minor   \n",
       "81745  109580095    HBB-LCR  0.815  0.069  C0271979  Thalassemia Intermedia   \n",
       "\n",
       "      diseaseType diseaseClass  diseaseSemanticType  score  EI  YearInitial  \\\n",
       "81741     disease      C15;C16  Disease or Syndrome    0.3 NaN          NaN   \n",
       "81742     disease      C15;C16  Disease or Syndrome    0.3 NaN          NaN   \n",
       "81743     disease      C15;C16  Disease or Syndrome    0.3 NaN          NaN   \n",
       "81744     disease      C15;C16  Disease or Syndrome    0.3 NaN          NaN   \n",
       "81745     disease      C15;C16  Disease or Syndrome    0.3 NaN          NaN   \n",
       "\n",
       "       YearFinal  NofPmids  NofSnps     source  \n",
       "81741        NaN         0        0  CTD_human  \n",
       "81742        NaN         0        0  CTD_human  \n",
       "81743        NaN         0        0  CTD_human  \n",
       "81744        NaN         0        0  CTD_human  \n",
       "81745        NaN         0        0  CTD_human  "
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_disease.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>DSI</th>\n",
       "      <th>DPI</th>\n",
       "      <th>diseaseId</th>\n",
       "      <th>diseaseName</th>\n",
       "      <th>diseaseType</th>\n",
       "      <th>diseaseClass</th>\n",
       "      <th>diseaseSemanticType</th>\n",
       "      <th>score</th>\n",
       "      <th>EI</th>\n",
       "      <th>YearInitial</th>\n",
       "      <th>YearFinal</th>\n",
       "      <th>NofPmids</th>\n",
       "      <th>NofSnps</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6822</th>\n",
       "      <td>790</td>\n",
       "      <td>CAD</td>\n",
       "      <td>0.557</td>\n",
       "      <td>0.655</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1979.0</td>\n",
       "      <td>1979.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14149</th>\n",
       "      <td>1723</td>\n",
       "      <td>DHODH</td>\n",
       "      <td>0.621</td>\n",
       "      <td>0.621</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1979.0</td>\n",
       "      <td>1979.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16472</th>\n",
       "      <td>2056</td>\n",
       "      <td>EPO</td>\n",
       "      <td>0.425</td>\n",
       "      <td>0.759</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23329</th>\n",
       "      <td>3040</td>\n",
       "      <td>HBA2</td>\n",
       "      <td>0.524</td>\n",
       "      <td>0.724</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.31</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1982.0</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23351</th>\n",
       "      <td>3043</td>\n",
       "      <td>HBB</td>\n",
       "      <td>0.513</td>\n",
       "      <td>0.793</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.60</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1979.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human;ORPHANET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23383</th>\n",
       "      <td>3047</td>\n",
       "      <td>HBG1</td>\n",
       "      <td>0.639</td>\n",
       "      <td>0.448</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.40</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1982.0</td>\n",
       "      <td>2010.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ORPHANET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23390</th>\n",
       "      <td>3048</td>\n",
       "      <td>HBG2</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.552</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ORPHANET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30758</th>\n",
       "      <td>3934</td>\n",
       "      <td>LCN2</td>\n",
       "      <td>0.469</td>\n",
       "      <td>0.793</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51059</th>\n",
       "      <td>7036</td>\n",
       "      <td>TFR2</td>\n",
       "      <td>0.594</td>\n",
       "      <td>0.517</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51080</th>\n",
       "      <td>7037</td>\n",
       "      <td>TFRC</td>\n",
       "      <td>0.471</td>\n",
       "      <td>0.759</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54111</th>\n",
       "      <td>7372</td>\n",
       "      <td>UMPS</td>\n",
       "      <td>0.598</td>\n",
       "      <td>0.483</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1979.0</td>\n",
       "      <td>1979.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62282</th>\n",
       "      <td>10661</td>\n",
       "      <td>KLF1</td>\n",
       "      <td>0.630</td>\n",
       "      <td>0.276</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.32</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2010.0</td>\n",
       "      <td>2013.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ORPHANET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71722</th>\n",
       "      <td>57817</td>\n",
       "      <td>HAMP</td>\n",
       "      <td>0.499</td>\n",
       "      <td>0.759</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81743</th>\n",
       "      <td>109580095</td>\n",
       "      <td>HBB-LCR</td>\n",
       "      <td>0.815</td>\n",
       "      <td>0.069</td>\n",
       "      <td>C0019025</td>\n",
       "      <td>Hemoglobin F Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C15;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          geneId geneSymbol    DSI    DPI diseaseId           diseaseName  \\\n",
       "6822         790        CAD  0.557  0.655  C0019025  Hemoglobin F Disease   \n",
       "14149       1723      DHODH  0.621  0.621  C0019025  Hemoglobin F Disease   \n",
       "16472       2056        EPO  0.425  0.759  C0019025  Hemoglobin F Disease   \n",
       "23329       3040       HBA2  0.524  0.724  C0019025  Hemoglobin F Disease   \n",
       "23351       3043        HBB  0.513  0.793  C0019025  Hemoglobin F Disease   \n",
       "23383       3047       HBG1  0.639  0.448  C0019025  Hemoglobin F Disease   \n",
       "23390       3048       HBG2  0.636  0.552  C0019025  Hemoglobin F Disease   \n",
       "30758       3934       LCN2  0.469  0.793  C0019025  Hemoglobin F Disease   \n",
       "51059       7036       TFR2  0.594  0.517  C0019025  Hemoglobin F Disease   \n",
       "51080       7037       TFRC  0.471  0.759  C0019025  Hemoglobin F Disease   \n",
       "54111       7372       UMPS  0.598  0.483  C0019025  Hemoglobin F Disease   \n",
       "62282      10661       KLF1  0.630  0.276  C0019025  Hemoglobin F Disease   \n",
       "71722      57817       HAMP  0.499  0.759  C0019025  Hemoglobin F Disease   \n",
       "81743  109580095    HBB-LCR  0.815  0.069  C0019025  Hemoglobin F Disease   \n",
       "\n",
       "      diseaseType diseaseClass  diseaseSemanticType  score   EI  YearInitial  \\\n",
       "6822      disease      C15;C16  Disease or Syndrome   0.30  NaN       1979.0   \n",
       "14149     disease      C15;C16  Disease or Syndrome   0.30  NaN       1979.0   \n",
       "16472     disease      C15;C16  Disease or Syndrome   0.30  NaN       2006.0   \n",
       "23329     disease      C15;C16  Disease or Syndrome   0.31  1.0       1982.0   \n",
       "23351     disease      C15;C16  Disease or Syndrome   0.60  1.0       1979.0   \n",
       "23383     disease      C15;C16  Disease or Syndrome   0.40  1.0       1982.0   \n",
       "23390     disease      C15;C16  Disease or Syndrome   0.30  NaN          NaN   \n",
       "30758     disease      C15;C16  Disease or Syndrome   0.30  NaN       2006.0   \n",
       "51059     disease      C15;C16  Disease or Syndrome   0.30  NaN       2006.0   \n",
       "51080     disease      C15;C16  Disease or Syndrome   0.30  NaN       2006.0   \n",
       "54111     disease      C15;C16  Disease or Syndrome   0.30  NaN       1979.0   \n",
       "62282     disease      C15;C16  Disease or Syndrome   0.32  1.0       2010.0   \n",
       "71722     disease      C15;C16  Disease or Syndrome   0.30  NaN       2006.0   \n",
       "81743     disease      C15;C16  Disease or Syndrome   0.30  NaN          NaN   \n",
       "\n",
       "       YearFinal  NofPmids  NofSnps              source  \n",
       "6822      1979.0         1        0           CTD_human  \n",
       "14149     1979.0         1        0           CTD_human  \n",
       "16472     2006.0         1        0           CTD_human  \n",
       "23329     2007.0         1        0           CTD_human  \n",
       "23351     2017.0         6        0  CTD_human;ORPHANET  \n",
       "23383     2010.0         0        0            ORPHANET  \n",
       "23390        NaN         0        0            ORPHANET  \n",
       "30758     2006.0         1        0           CTD_human  \n",
       "51059     2006.0         1        0           CTD_human  \n",
       "51080     2006.0         1        0           CTD_human  \n",
       "54111     1979.0         1        0           CTD_human  \n",
       "62282     2013.0         0        0            ORPHANET  \n",
       "71722     2006.0         1        0           CTD_human  \n",
       "81743        NaN         0        0           CTD_human  "
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_disease[gene_disease.diseaseId == 'C0019025']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(9413, 10370)"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(gene_disease.geneId.unique()), len(gene_disease.diseaseId.unique())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Creating a table of associated diseases for every  mutated gene of the patient "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "metabolic_filename = 'shortened_metabolic.csv'\n",
    "pediatric_filename = 'shortened_Pediatric.csv'\n",
    "carrier_filename = 'shortened_Carrier.csv'\n",
    "cardioneuro_filename = 'shortened_CardioNeuro.csv'\n",
    "exome_filename = 'shortened_Exome.csv'\n",
    "\n",
    "\n",
    "metabolic_var_genes = pd.read_csv(metabolic_filename)\n",
    "pediatric_var_genes = pd.read_csv(pediatric_filename)\n",
    "carrier_var_genes = pd.read_csv(carrier_filename)\n",
    "cardioneuro_var_genes = pd.read_csv(cardioneuro_filename)\n",
    "exome_var_genes = pd.read_csv(exome_filename)\n",
    "\n",
    "metabolic_diseases = pd.DataFrame()\n",
    "pediatric_diseases = pd.DataFrame()\n",
    "carrier_diseases = pd.DataFrame()\n",
    "cardioneuro_diseases = pd.DataFrame()\n",
    "exome_diseases = pd.DataFrame()\n",
    "\n",
    "for variant in metabolic_var_genes.itertuples():\n",
    "    metabolic_diseases = metabolic_diseases.append(gene_disease[gene_disease.geneSymbol == variant.gene_name])\n",
    "\n",
    "for variant in pediatric_var_genes.itertuples():\n",
    "    pediatric_diseases = pediatric_diseases.append(gene_disease[gene_disease.geneSymbol == variant.gene_name])\n",
    "\n",
    "for variant in carrier_var_genes.itertuples():\n",
    "    carrier_diseases = carrier_diseases.append(gene_disease[gene_disease.geneSymbol == variant.gene_name])\n",
    "\n",
    "for variant in cardioneuro_var_genes.itertuples():\n",
    "    cardioneuro_diseases = cardioneuro_diseases.append(gene_disease[gene_disease.geneSymbol == variant.gene_name])\n",
    "\n",
    "for variant in exome_var_genes.itertuples():\n",
    "    exome_diseases = exome_diseases.append(gene_disease[gene_disease.geneSymbol == variant.gene_name])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Unnamed: 0, CHROM, start, end, strand, gene_id, transcript_id, gene_type, gene_status, gene_name, transcript_type, transcript_status, transcript_name, POS, REF, ALT, FILTER]\n",
       "Index: []"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exome_var_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 616 entries, 49215 to 50434\n",
      "Data columns (total 16 columns):\n",
      "geneId                 616 non-null int64\n",
      "geneSymbol             616 non-null object\n",
      "DSI                    616 non-null float64\n",
      "DPI                    616 non-null float64\n",
      "diseaseId              616 non-null object\n",
      "diseaseName            616 non-null object\n",
      "diseaseType            616 non-null object\n",
      "diseaseClass           554 non-null object\n",
      "diseaseSemanticType    616 non-null object\n",
      "score                  616 non-null float64\n",
      "EI                     183 non-null float64\n",
      "YearInitial            532 non-null float64\n",
      "YearFinal              532 non-null float64\n",
      "NofPmids               616 non-null int64\n",
      "NofSnps                616 non-null int64\n",
      "source                 616 non-null object\n",
      "dtypes: float64(6), int64(3), object(7)\n",
      "memory usage: 81.8+ KB\n"
     ]
    }
   ],
   "source": [
    "metabolic_diseases.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index: 0 entries\n",
      "Empty DataFrame"
     ]
    }
   ],
   "source": [
    "exome_diseases.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>DSI</th>\n",
       "      <th>DPI</th>\n",
       "      <th>diseaseId</th>\n",
       "      <th>diseaseName</th>\n",
       "      <th>diseaseType</th>\n",
       "      <th>diseaseClass</th>\n",
       "      <th>diseaseSemanticType</th>\n",
       "      <th>score</th>\n",
       "      <th>EI</th>\n",
       "      <th>YearInitial</th>\n",
       "      <th>YearFinal</th>\n",
       "      <th>NofPmids</th>\n",
       "      <th>NofSnps</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>75346</th>\n",
       "      <td>90529</td>\n",
       "      <td>STPG1</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.069</td>\n",
       "      <td>C0006142</td>\n",
       "      <td>Malignant neoplasm of breast</td>\n",
       "      <td>disease</td>\n",
       "      <td>C04;C17</td>\n",
       "      <td>Neoplastic Process</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>UNIPROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73351</th>\n",
       "      <td>79753</td>\n",
       "      <td>SNIP1</td>\n",
       "      <td>0.685</td>\n",
       "      <td>0.345</td>\n",
       "      <td>C3281055</td>\n",
       "      <td>PSYCHOMOTOR RETARDATION, EPILEPSY, AND CRANIOF...</td>\n",
       "      <td>disease</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2012.0</td>\n",
       "      <td>2012.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>UNIPROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73352</th>\n",
       "      <td>79753</td>\n",
       "      <td>SNIP1</td>\n",
       "      <td>0.685</td>\n",
       "      <td>0.345</td>\n",
       "      <td>C3714756</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>group</td>\n",
       "      <td>C10;C23;F01;F03</td>\n",
       "      <td>Mental or Behavioral Dysfunction</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2012.0</td>\n",
       "      <td>2012.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>GENOMICS_ENGLAND</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44983</th>\n",
       "      <td>6121</td>\n",
       "      <td>RPE65</td>\n",
       "      <td>0.479</td>\n",
       "      <td>0.759</td>\n",
       "      <td>C0015398</td>\n",
       "      <td>Eye Diseases, Hereditary</td>\n",
       "      <td>group</td>\n",
       "      <td>C11;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44984</th>\n",
       "      <td>6121</td>\n",
       "      <td>RPE65</td>\n",
       "      <td>0.479</td>\n",
       "      <td>0.759</td>\n",
       "      <td>C0035304</td>\n",
       "      <td>Retinal Degeneration</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>C11</td>\n",
       "      <td>Pathologic Function</td>\n",
       "      <td>0.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2014.0</td>\n",
       "      <td>2015.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       geneId geneSymbol    DSI    DPI diseaseId  \\\n",
       "75346   90529      STPG1  1.000  0.069  C0006142   \n",
       "73351   79753      SNIP1  0.685  0.345  C3281055   \n",
       "73352   79753      SNIP1  0.685  0.345  C3714756   \n",
       "44983    6121      RPE65  0.479  0.759  C0015398   \n",
       "44984    6121      RPE65  0.479  0.759  C0035304   \n",
       "\n",
       "                                             diseaseName diseaseType  \\\n",
       "75346                       Malignant neoplasm of breast     disease   \n",
       "73351  PSYCHOMOTOR RETARDATION, EPILEPSY, AND CRANIOF...     disease   \n",
       "73352                            Intellectual Disability       group   \n",
       "44983                           Eye Diseases, Hereditary       group   \n",
       "44984                               Retinal Degeneration   phenotype   \n",
       "\n",
       "          diseaseClass               diseaseSemanticType  score  EI  \\\n",
       "75346          C04;C17                Neoplastic Process    0.3 NaN   \n",
       "73351              NaN               Disease or Syndrome    0.4 NaN   \n",
       "73352  C10;C23;F01;F03  Mental or Behavioral Dysfunction    0.3 NaN   \n",
       "44983          C11;C16               Disease or Syndrome    0.3 NaN   \n",
       "44984              C11               Pathologic Function    0.4 NaN   \n",
       "\n",
       "       YearInitial  YearFinal  NofPmids  NofSnps            source  \n",
       "75346          NaN        NaN         0        0           UNIPROT  \n",
       "73351       2012.0     2012.0         1        1           UNIPROT  \n",
       "73352       2012.0     2012.0         1        0  GENOMICS_ENGLAND  \n",
       "44983       2007.0     2007.0         1        0         CTD_human  \n",
       "44984       2014.0     2015.0         1        1         CTD_human  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pediatric_diseases.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>DSI</th>\n",
       "      <th>DPI</th>\n",
       "      <th>diseaseId</th>\n",
       "      <th>diseaseName</th>\n",
       "      <th>diseaseType</th>\n",
       "      <th>diseaseClass</th>\n",
       "      <th>diseaseSemanticType</th>\n",
       "      <th>score</th>\n",
       "      <th>EI</th>\n",
       "      <th>YearInitial</th>\n",
       "      <th>YearFinal</th>\n",
       "      <th>NofPmids</th>\n",
       "      <th>NofSnps</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>75931</th>\n",
       "      <td>114034</td>\n",
       "      <td>TOE1</td>\n",
       "      <td>0.701</td>\n",
       "      <td>0.414</td>\n",
       "      <td>C0036875</td>\n",
       "      <td>Disorders of Sex Development</td>\n",
       "      <td>group</td>\n",
       "      <td>C12;C13;C16;C19</td>\n",
       "      <td>Congenital Abnormality</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>GENOMICS_ENGLAND</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75932</th>\n",
       "      <td>114034</td>\n",
       "      <td>TOE1</td>\n",
       "      <td>0.701</td>\n",
       "      <td>0.414</td>\n",
       "      <td>C0266468</td>\n",
       "      <td>Congenital pontocerebellar hypoplasia</td>\n",
       "      <td>disease</td>\n",
       "      <td>C10</td>\n",
       "      <td>Congenital Abnormality</td>\n",
       "      <td>0.31</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75933</th>\n",
       "      <td>114034</td>\n",
       "      <td>TOE1</td>\n",
       "      <td>0.701</td>\n",
       "      <td>0.414</td>\n",
       "      <td>C0266470</td>\n",
       "      <td>Cerebellar Hypoplasia</td>\n",
       "      <td>disease</td>\n",
       "      <td>C10;C16;F03</td>\n",
       "      <td>Congenital Abnormality</td>\n",
       "      <td>0.40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>GENOMICS_ENGLAND</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75934</th>\n",
       "      <td>114034</td>\n",
       "      <td>TOE1</td>\n",
       "      <td>0.701</td>\n",
       "      <td>0.414</td>\n",
       "      <td>C3554226</td>\n",
       "      <td>Congenital pontocerebellar hypoplasia type 7</td>\n",
       "      <td>disease</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.70</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>CTD_human;ORPHANET;UNIPROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75935</th>\n",
       "      <td>114034</td>\n",
       "      <td>TOE1</td>\n",
       "      <td>0.701</td>\n",
       "      <td>0.414</td>\n",
       "      <td>C3714756</td>\n",
       "      <td>Intellectual Disability</td>\n",
       "      <td>group</td>\n",
       "      <td>C10;C23;F01;F03</td>\n",
       "      <td>Mental or Behavioral Dysfunction</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>GENOMICS_ENGLAND</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       geneId geneSymbol    DSI    DPI diseaseId  \\\n",
       "75931  114034       TOE1  0.701  0.414  C0036875   \n",
       "75932  114034       TOE1  0.701  0.414  C0266468   \n",
       "75933  114034       TOE1  0.701  0.414  C0266470   \n",
       "75934  114034       TOE1  0.701  0.414  C3554226   \n",
       "75935  114034       TOE1  0.701  0.414  C3714756   \n",
       "\n",
       "                                        diseaseName diseaseType  \\\n",
       "75931                  Disorders of Sex Development       group   \n",
       "75932         Congenital pontocerebellar hypoplasia     disease   \n",
       "75933                         Cerebellar Hypoplasia     disease   \n",
       "75934  Congenital pontocerebellar hypoplasia type 7     disease   \n",
       "75935                       Intellectual Disability       group   \n",
       "\n",
       "          diseaseClass               diseaseSemanticType  score   EI  \\\n",
       "75931  C12;C13;C16;C19            Congenital Abnormality   0.30  NaN   \n",
       "75932              C10            Congenital Abnormality   0.31  1.0   \n",
       "75933      C10;C16;F03            Congenital Abnormality   0.40  NaN   \n",
       "75934              NaN               Disease or Syndrome   0.70  NaN   \n",
       "75935  C10;C23;F01;F03  Mental or Behavioral Dysfunction   0.30  NaN   \n",
       "\n",
       "       YearInitial  YearFinal  NofPmids  NofSnps                      source  \n",
       "75931       2017.0     2017.0         1        0            GENOMICS_ENGLAND  \n",
       "75932       2017.0     2017.0         1        0                   CTD_human  \n",
       "75933       2017.0     2017.0         1        0            GENOMICS_ENGLAND  \n",
       "75934       2017.0     2017.0         1        8  CTD_human;ORPHANET;UNIPROT  \n",
       "75935       2017.0     2017.0         1        0            GENOMICS_ENGLAND  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "carrier_diseases.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>DSI</th>\n",
       "      <th>DPI</th>\n",
       "      <th>diseaseId</th>\n",
       "      <th>diseaseName</th>\n",
       "      <th>diseaseType</th>\n",
       "      <th>diseaseClass</th>\n",
       "      <th>diseaseSemanticType</th>\n",
       "      <th>score</th>\n",
       "      <th>EI</th>\n",
       "      <th>YearInitial</th>\n",
       "      <th>YearFinal</th>\n",
       "      <th>NofPmids</th>\n",
       "      <th>NofSnps</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24581</th>\n",
       "      <td>3208</td>\n",
       "      <td>HPCA</td>\n",
       "      <td>0.735</td>\n",
       "      <td>0.345</td>\n",
       "      <td>C1857093</td>\n",
       "      <td>DYSTONIA 2, TORSION, AUTOSOMAL RECESSIVE (diso...</td>\n",
       "      <td>disease</td>\n",
       "      <td>C10;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.70</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1981.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>CTD_human;ORPHANET;UNIPROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35781</th>\n",
       "      <td>4803</td>\n",
       "      <td>NGF</td>\n",
       "      <td>0.426</td>\n",
       "      <td>0.862</td>\n",
       "      <td>C0002622</td>\n",
       "      <td>Amnesia</td>\n",
       "      <td>disease</td>\n",
       "      <td>C10;C23;F01;F03</td>\n",
       "      <td>Mental or Behavioral Dysfunction</td>\n",
       "      <td>0.31</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1995.0</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35782</th>\n",
       "      <td>4803</td>\n",
       "      <td>NGF</td>\n",
       "      <td>0.426</td>\n",
       "      <td>0.862</td>\n",
       "      <td>C0002768</td>\n",
       "      <td>Congenital Pain Insensitivity</td>\n",
       "      <td>disease</td>\n",
       "      <td>C10;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.32</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>2013.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ORPHANET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35783</th>\n",
       "      <td>4803</td>\n",
       "      <td>NGF</td>\n",
       "      <td>0.426</td>\n",
       "      <td>0.862</td>\n",
       "      <td>C0007959</td>\n",
       "      <td>Charcot-Marie-Tooth Disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>C10;C16</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>GENOMICS_ENGLAND</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35784</th>\n",
       "      <td>4803</td>\n",
       "      <td>NGF</td>\n",
       "      <td>0.426</td>\n",
       "      <td>0.862</td>\n",
       "      <td>C0009171</td>\n",
       "      <td>Cocaine Abuse</td>\n",
       "      <td>disease</td>\n",
       "      <td>C25;F03</td>\n",
       "      <td>Mental or Behavioral Dysfunction</td>\n",
       "      <td>0.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008.0</td>\n",
       "      <td>2008.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTD_human</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       geneId geneSymbol    DSI    DPI diseaseId  \\\n",
       "24581    3208       HPCA  0.735  0.345  C1857093   \n",
       "35781    4803        NGF  0.426  0.862  C0002622   \n",
       "35782    4803        NGF  0.426  0.862  C0002768   \n",
       "35783    4803        NGF  0.426  0.862  C0007959   \n",
       "35784    4803        NGF  0.426  0.862  C0009171   \n",
       "\n",
       "                                             diseaseName diseaseType  \\\n",
       "24581  DYSTONIA 2, TORSION, AUTOSOMAL RECESSIVE (diso...     disease   \n",
       "35781                                            Amnesia     disease   \n",
       "35782                      Congenital Pain Insensitivity     disease   \n",
       "35783                        Charcot-Marie-Tooth Disease     disease   \n",
       "35784                                      Cocaine Abuse     disease   \n",
       "\n",
       "          diseaseClass               diseaseSemanticType  score   EI  \\\n",
       "24581          C10;C16               Disease or Syndrome   0.70  NaN   \n",
       "35781  C10;C23;F01;F03  Mental or Behavioral Dysfunction   0.31  1.0   \n",
       "35782          C10;C16               Disease or Syndrome   0.32  1.0   \n",
       "35783          C10;C16               Disease or Syndrome   0.30  NaN   \n",
       "35784          C25;F03  Mental or Behavioral Dysfunction   0.30  NaN   \n",
       "\n",
       "       YearInitial  YearFinal  NofPmids  NofSnps                      source  \n",
       "24581       1981.0     2017.0         2        3  CTD_human;ORPHANET;UNIPROT  \n",
       "35781       1995.0     2009.0         2        0                   CTD_human  \n",
       "35782       2009.0     2013.0         0        0                    ORPHANET  \n",
       "35783          NaN        NaN         0        0            GENOMICS_ENGLAND  \n",
       "35784       2008.0     2008.0         1        0                   CTD_human  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cardioneuro_diseases.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_diseases = pd.DataFrame()\n",
    "all_diseases = all_diseases.append(metabolic_diseases)\n",
    "all_diseases = all_diseases.append(pediatric_diseases)\n",
    "all_diseases = all_diseases.append(carrier_diseases)\n",
    "all_diseases = all_diseases.append(cardioneuro_diseases)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 3494 entries, 49215 to 50434\n",
      "Data columns (total 16 columns):\n",
      "geneId                 3494 non-null int64\n",
      "geneSymbol             3494 non-null object\n",
      "DSI                    3491 non-null float64\n",
      "DPI                    3491 non-null float64\n",
      "diseaseId              3494 non-null object\n",
      "diseaseName            3494 non-null object\n",
      "diseaseType            3494 non-null object\n",
      "diseaseClass           3196 non-null object\n",
      "diseaseSemanticType    3494 non-null object\n",
      "score                  3494 non-null float64\n",
      "EI                     1044 non-null float64\n",
      "YearInitial            3083 non-null float64\n",
      "YearFinal              3083 non-null float64\n",
      "NofPmids               3494 non-null int64\n",
      "NofSnps                3494 non-null int64\n",
      "source                 3494 non-null object\n",
      "dtypes: float64(6), int64(3), object(7)\n",
      "memory usage: 464.0+ KB\n"
     ]
    }
   ],
   "source": [
    "all_diseases.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_diseases.to_csv('diseases_patient.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 81746 entries, 0 to 81745\n",
      "Data columns (total 16 columns):\n",
      "geneId                 81746 non-null int64\n",
      "geneSymbol             81746 non-null object\n",
      "DSI                    81620 non-null float64\n",
      "DPI                    81598 non-null float64\n",
      "diseaseId              81746 non-null object\n",
      "diseaseName            81746 non-null object\n",
      "diseaseType            81746 non-null object\n",
      "diseaseClass           75443 non-null object\n",
      "diseaseSemanticType    81746 non-null object\n",
      "score                  81746 non-null float64\n",
      "EI                     22862 non-null float64\n",
      "YearInitial            73123 non-null float64\n",
      "YearFinal              73123 non-null float64\n",
      "NofPmids               81746 non-null int64\n",
      "NofSnps                81746 non-null int64\n",
      "source                 81746 non-null object\n",
      "dtypes: float64(6), int64(3), object(7)\n",
      "memory usage: 10.0+ MB\n"
     ]
    }
   ],
   "source": [
    "gene_disease.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_genes = gene_disease.geneSymbol.unique()\n",
    "unique_diseases = gene_disease.diseaseId.unique()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = pd.DataFrame(np.zeros(( len(unique_diseases), len(unique_genes))), columns=unique_genes, index=unique_diseases)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "for row in gene_disease.itertuples():\n",
    "    gene = row.geneSymbol\n",
    "    disease = row.diseaseId\n",
    "    score = row.score\n",
    "    data[gene].loc[disease] = score\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>NAT1</th>\n",
       "      <th>NAT2</th>\n",
       "      <th>SERPINA3</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AANAT</th>\n",
       "      <th>AARS</th>\n",
       "      <th>ABAT</th>\n",
       "      <th>ABCA1</th>\n",
       "      <th>...</th>\n",
       "      <th>MIR4433B</th>\n",
       "      <th>MIR7977</th>\n",
       "      <th>MIR1273H</th>\n",
       "      <th>MIR8061</th>\n",
       "      <th>MIR6741</th>\n",
       "      <th>MIR6803</th>\n",
       "      <th>SCA37</th>\n",
       "      <th>H19-ICR</th>\n",
       "      <th>DHS6S1</th>\n",
       "      <th>HBB-LCR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C0019209</th>\n",
       "      <td>0.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C0036341</th>\n",
       "      <td>0.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.31</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C0002395</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C0007102</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C0009375</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 9411 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          A1BG  A2M  NAT1  NAT2  SERPINA3  AADAC  AANAT  AARS  ABAT  ABCA1  \\\n",
       "C0019209   0.3  0.0   0.0   0.0       0.0    0.0    0.0   0.0   0.0   0.00   \n",
       "C0036341   0.3  0.0   0.0   0.0       0.0    0.0    0.0   0.0   0.0   0.31   \n",
       "C0002395   0.0  0.4   0.0   0.0       0.0    0.0    0.0   0.0   0.0   0.00   \n",
       "C0007102   0.0  0.3   0.0   0.0       0.0    0.0    0.0   0.0   0.0   0.00   \n",
       "C0009375   0.0  0.3   0.0   0.0       0.0    0.0    0.0   0.0   0.0   0.00   \n",
       "\n",
       "           ...     MIR4433B  MIR7977  MIR1273H  MIR8061  MIR6741  MIR6803  \\\n",
       "C0019209   ...          0.0      0.0       0.0      0.0      0.0      0.0   \n",
       "C0036341   ...          0.0      0.0       0.0      0.0      0.0      0.0   \n",
       "C0002395   ...          0.0      0.0       0.0      0.0      0.0      0.0   \n",
       "C0007102   ...          0.0      0.0       0.0      0.0      0.0      0.0   \n",
       "C0009375   ...          0.0      0.0       0.0      0.0      0.0      0.0   \n",
       "\n",
       "          SCA37  H19-ICR  DHS6S1  HBB-LCR  \n",
       "C0019209    0.0      0.0     0.0      0.0  \n",
       "C0036341    0.0      0.0     0.0      0.0  \n",
       "C0002395    0.0      0.0     0.0      0.0  \n",
       "C0007102    0.0      0.0     0.0      0.0  \n",
       "C0009375    0.0      0.0     0.0      0.0  \n",
       "\n",
       "[5 rows x 9411 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "C0019209    0.00\n",
       "C0036341    0.00\n",
       "C0002395    0.00\n",
       "C0007102    0.32\n",
       "C0009375    0.30\n",
       "Name: TYMP, dtype: float64"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.TYMP.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_csv('matrixA.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = pd.DataFrame(np.zeros((1, len(unique_genes))), columns=unique_genes, index=[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "for row in all_diseases.itertuples():\n",
    "    gene = row.geneSymbol\n",
    "    g[gene].loc[0] = 1.\n",
    "    #print(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.018170226330889386"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.mean(g.loc[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "G = np.array(g.loc[0])\n",
    "A = data.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = np.dot(A, G)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_d = pd.DataFrame(columns=unique_diseases)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_d.loc['a'] = d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C0019209</th>\n",
       "      <th>C0036341</th>\n",
       "      <th>C0002395</th>\n",
       "      <th>C0007102</th>\n",
       "      <th>C0009375</th>\n",
       "      <th>C0011265</th>\n",
       "      <th>C0011570</th>\n",
       "      <th>C0011581</th>\n",
       "      <th>C0019202</th>\n",
       "      <th>C0022660</th>\n",
       "      <th>...</th>\n",
       "      <th>C1859452</th>\n",
       "      <th>C4319565</th>\n",
       "      <th>C2676281</th>\n",
       "      <th>C2936793</th>\n",
       "      <th>C3888208</th>\n",
       "      <th>C3150680</th>\n",
       "      <th>C2931662</th>\n",
       "      <th>C3151343</th>\n",
       "      <th>C3281236</th>\n",
       "      <th>C0730294</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>a</th>\n",
       "      <td>0.3</td>\n",
       "      <td>5.09</td>\n",
       "      <td>1.14</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>1.07</td>\n",
       "      <td>2.09</td>\n",
       "      <td>2.4</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 10370 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   C0019209  C0036341  C0002395  C0007102  C0009375  C0011265  C0011570  \\\n",
       "a       0.3      5.09      1.14       0.3       0.3      1.07      2.09   \n",
       "\n",
       "   C0011581  C0019202  C0022660    ...     C1859452  C4319565  C2676281  \\\n",
       "a       2.4      0.32       0.0    ...          0.0       0.0       0.0   \n",
       "\n",
       "   C2936793  C3888208  C3150680  C2931662  C3151343  C3281236  C0730294  \n",
       "a       0.0       0.0       0.0       0.0       0.0       0.0       0.0  \n",
       "\n",
       "[1 rows x 10370 columns]"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_d"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## For each part of the genome I create vector g and d:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "g_metabolic = pd.DataFrame(np.zeros((1, len(unique_genes))), columns=unique_genes, index=[0])\n",
    "g_pediatric = pd.DataFrame(np.zeros((1, len(unique_genes))), columns=unique_genes, index=[0])\n",
    "g_carrier = pd.DataFrame(np.zeros((1, len(unique_genes))), columns=unique_genes, index=[0])\n",
    "g_cardioneuro = pd.DataFrame(np.zeros((1, len(unique_genes))), columns=unique_genes, index=[0])\n",
    "g = pd.DataFrame(np.zeros((1, len(unique_genes))), columns=unique_genes, index=[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [],
   "source": [
    "for row in all_diseases.itertuples():\n",
    "    gene = row.geneSymbol\n",
    "    g[gene].loc[0] = 1.\n",
    "    #print(gene)\n",
    "\n",
    "for row in metabolic_diseases.itertuples():\n",
    "    gene = row.geneSymbol\n",
    "    g_metabolic[gene].loc[0] = 1.\n",
    "    #print(gene)\n",
    "\n",
    "for row in pediatric_diseases.itertuples():\n",
    "    gene = row.geneSymbol\n",
    "    g_pediatric[gene].loc[0] = 1.\n",
    "    #print(gene)\n",
    "    \n",
    "for row in carrier_diseases.itertuples():\n",
    "    gene = row.geneSymbol\n",
    "    g_carrier[gene].loc[0] = 1.\n",
    "    #print(gene)\n",
    "    \n",
    "for row in cardioneuro_diseases.itertuples():\n",
    "    gene = row.geneSymbol\n",
    "    g_cardioneuro[gene].loc[0] = 1.\n",
    "    #print(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "G_metabolic = np.array(g_metabolic.loc[0])\n",
    "G_pediatric = np.array(g_pediatric.loc[0])\n",
    "G_carrier = np.array(g_carrier.loc[0])\n",
    "G_cardioneuro = np.array(g_cardioneuro.loc[0])\n",
    "G = np.array(g.loc[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = np.dot(A, G)\n",
    "d_metabolic = np.dot(A, G_metabolic)\n",
    "d_pediatric = np.dot(A, G_pediatric)\n",
    "d_carrier = np.dot(A, G_carrier)\n",
    "d_cardioneuro = np.dot(A, G_cardioneuro)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [],
   "source": [
    "d_df = pd.DataFrame(columns=unique_diseases)\n",
    "d_df.loc['a'] = d\n",
    "\n",
    "d_metabolic_df = pd.DataFrame(columns=unique_diseases)\n",
    "d_metabolic_df.loc['a'] = d_metabolic\n",
    "\n",
    "d_pediatric_df = pd.DataFrame(columns=unique_diseases)\n",
    "d_pediatric_df.loc['a'] = d_pediatric\n",
    "\n",
    "d_carrier_df = pd.DataFrame(columns=unique_diseases)\n",
    "d_carrier_df.loc['a'] = d_carrier\n",
    "\n",
    "d_cardioneuro_df = pd.DataFrame(columns=unique_diseases)\n",
    "d_cardioneuro_df.loc['a'] = d_cardioneuro"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C0019209</th>\n",
       "      <th>C0036341</th>\n",
       "      <th>C0002395</th>\n",
       "      <th>C0007102</th>\n",
       "      <th>C0009375</th>\n",
       "      <th>C0011265</th>\n",
       "      <th>C0011570</th>\n",
       "      <th>C0011581</th>\n",
       "      <th>C0019202</th>\n",
       "      <th>C0022660</th>\n",
       "      <th>...</th>\n",
       "      <th>C1859452</th>\n",
       "      <th>C4319565</th>\n",
       "      <th>C2676281</th>\n",
       "      <th>C2936793</th>\n",
       "      <th>C3888208</th>\n",
       "      <th>C3150680</th>\n",
       "      <th>C2931662</th>\n",
       "      <th>C3151343</th>\n",
       "      <th>C3281236</th>\n",
       "      <th>C0730294</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>a</th>\n",
       "      <td>0.3</td>\n",
       "      <td>5.09</td>\n",
       "      <td>1.14</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>1.07</td>\n",
       "      <td>2.09</td>\n",
       "      <td>2.4</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 10370 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   C0019209  C0036341  C0002395  C0007102  C0009375  C0011265  C0011570  \\\n",
       "a       0.3      5.09      1.14       0.3       0.3      1.07      2.09   \n",
       "\n",
       "   C0011581  C0019202  C0022660    ...     C1859452  C4319565  C2676281  \\\n",
       "a       2.4      0.32       0.0    ...          0.0       0.0       0.0   \n",
       "\n",
       "   C2936793  C3888208  C3150680  C2931662  C3151343  C3281236  C0730294  \n",
       "a       0.0       0.0       0.0       0.0       0.0       0.0       0.0  \n",
       "\n",
       "[1 rows x 10370 columns]"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C0019209</th>\n",
       "      <th>C0036341</th>\n",
       "      <th>C0002395</th>\n",
       "      <th>C0007102</th>\n",
       "      <th>C0009375</th>\n",
       "      <th>C0011265</th>\n",
       "      <th>C0011570</th>\n",
       "      <th>C0011581</th>\n",
       "      <th>C0019202</th>\n",
       "      <th>C0022660</th>\n",
       "      <th>...</th>\n",
       "      <th>C1859452</th>\n",
       "      <th>C4319565</th>\n",
       "      <th>C2676281</th>\n",
       "      <th>C2936793</th>\n",
       "      <th>C3888208</th>\n",
       "      <th>C3150680</th>\n",
       "      <th>C2931662</th>\n",
       "      <th>C3151343</th>\n",
       "      <th>C3281236</th>\n",
       "      <th>C0730294</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>a</th>\n",
       "      <td>0.3</td>\n",
       "      <td>2.25</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.8</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 10370 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   C0019209  C0036341  C0002395  C0007102  C0009375  C0011265  C0011570  \\\n",
       "a       0.3      2.25       0.0       0.0       0.0       0.0       0.8   \n",
       "\n",
       "   C0011581  C0019202  C0022660    ...     C1859452  C4319565  C2676281  \\\n",
       "a      0.91       0.0       0.0    ...          0.0       0.0       0.0   \n",
       "\n",
       "   C2936793  C3888208  C3150680  C2931662  C3151343  C3281236  C0730294  \n",
       "a       0.0       0.0       0.0       0.0       0.0       0.0       0.0  \n",
       "\n",
       "[1 rows x 10370 columns]"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d_metabolic_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C0019209</th>\n",
       "      <th>C0036341</th>\n",
       "      <th>C0002395</th>\n",
       "      <th>C0007102</th>\n",
       "      <th>C0009375</th>\n",
       "      <th>C0011265</th>\n",
       "      <th>C0011570</th>\n",
       "      <th>C0011581</th>\n",
       "      <th>C0019202</th>\n",
       "      <th>C0022660</th>\n",
       "      <th>...</th>\n",
       "      <th>C1859452</th>\n",
       "      <th>C4319565</th>\n",
       "      <th>C2676281</th>\n",
       "      <th>C2936793</th>\n",
       "      <th>C3888208</th>\n",
       "      <th>C3150680</th>\n",
       "      <th>C2931662</th>\n",
       "      <th>C3151343</th>\n",
       "      <th>C3281236</th>\n",
       "      <th>C0730294</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>a</th>\n",
       "      <td>0.3</td>\n",
       "      <td>3.22</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.8</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 10370 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   C0019209  C0036341  C0002395  C0007102  C0009375  C0011265  C0011570  \\\n",
       "a       0.3      3.22      0.34       0.0       0.0       0.3       0.8   \n",
       "\n",
       "   C0011581  C0019202  C0022660    ...     C1859452  C4319565  C2676281  \\\n",
       "a      0.91       0.0       0.0    ...          0.0       0.0       0.0   \n",
       "\n",
       "   C2936793  C3888208  C3150680  C2931662  C3151343  C3281236  C0730294  \n",
       "a       0.0       0.0       0.0       0.0       0.0       0.0       0.0  \n",
       "\n",
       "[1 rows x 10370 columns]"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d_pediatric_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C0019209</th>\n",
       "      <th>C0036341</th>\n",
       "      <th>C0002395</th>\n",
       "      <th>C0007102</th>\n",
       "      <th>C0009375</th>\n",
       "      <th>C0011265</th>\n",
       "      <th>C0011570</th>\n",
       "      <th>C0011581</th>\n",
       "      <th>C0019202</th>\n",
       "      <th>C0022660</th>\n",
       "      <th>...</th>\n",
       "      <th>C1859452</th>\n",
       "      <th>C4319565</th>\n",
       "      <th>C2676281</th>\n",
       "      <th>C2936793</th>\n",
       "      <th>C3888208</th>\n",
       "      <th>C3150680</th>\n",
       "      <th>C2931662</th>\n",
       "      <th>C3151343</th>\n",
       "      <th>C3281236</th>\n",
       "      <th>C0730294</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>a</th>\n",
       "      <td>0.3</td>\n",
       "      <td>3.01</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.8</td>\n",
       "      <td>1.31</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 10370 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   C0019209  C0036341  C0002395  C0007102  C0009375  C0011265  C0011570  \\\n",
       "a       0.3      3.01       0.4       0.3       0.3       0.4       0.8   \n",
       "\n",
       "   C0011581  C0019202  C0022660    ...     C1859452  C4319565  C2676281  \\\n",
       "a      1.31      0.32       0.0    ...          0.0       0.0       0.0   \n",
       "\n",
       "   C2936793  C3888208  C3150680  C2931662  C3151343  C3281236  C0730294  \n",
       "a       0.0       0.0       0.0       0.0       0.0       0.0       0.0  \n",
       "\n",
       "[1 rows x 10370 columns]"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d_carrier_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C0019209</th>\n",
       "      <th>C0036341</th>\n",
       "      <th>C0002395</th>\n",
       "      <th>C0007102</th>\n",
       "      <th>C0009375</th>\n",
       "      <th>C0011265</th>\n",
       "      <th>C0011570</th>\n",
       "      <th>C0011581</th>\n",
       "      <th>C0019202</th>\n",
       "      <th>C0022660</th>\n",
       "      <th>...</th>\n",
       "      <th>C1859452</th>\n",
       "      <th>C4319565</th>\n",
       "      <th>C2676281</th>\n",
       "      <th>C2936793</th>\n",
       "      <th>C3888208</th>\n",
       "      <th>C3150680</th>\n",
       "      <th>C2931662</th>\n",
       "      <th>C3151343</th>\n",
       "      <th>C3281236</th>\n",
       "      <th>C0730294</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>a</th>\n",
       "      <td>0.3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.8</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.77</td>\n",
       "      <td>2.09</td>\n",
       "      <td>2.4</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 10370 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   C0019209  C0036341  C0002395  C0007102  C0009375  C0011265  C0011570  \\\n",
       "a       0.3       3.0       0.8       0.0       0.0      0.77      2.09   \n",
       "\n",
       "   C0011581  C0019202  C0022660    ...     C1859452  C4319565  C2676281  \\\n",
       "a       2.4      0.32       0.0    ...          0.0       0.0       0.0   \n",
       "\n",
       "   C2936793  C3888208  C3150680  C2931662  C3151343  C3281236  C0730294  \n",
       "a       0.0       0.0       0.0       0.0       0.0       0.0       0.0  \n",
       "\n",
       "[1 rows x 10370 columns]"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d_cardioneuro_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Intellectual Disability\n",
      "1 Epileptic encephalopathy\n",
      "2 Malignant neoplasm of breast\n",
      "3 Malignant neoplasm of prostate\n",
      "4 Charcot-Marie-Tooth Disease\n",
      "5 Prostatic Neoplasms\n",
      "6 Schizophrenia\n",
      "7 Liver Cirrhosis, Experimental\n",
      "8 Colorectal Cancer\n",
      "9 Breast Carcinoma\n"
     ]
    }
   ],
   "source": [
    "top10_all = open('top10_all.txt', 'w')\n",
    "\n",
    "top10_all_ids = d_df.sort_values('a',axis=1,ascending=False).columns[:10]\n",
    "for num, i in enumerate(top10_all_ids):\n",
    "    print(num, all_diseases[all_diseases.diseaseId == i].diseaseName.values[0])\n",
    "    top10_all.write('{} {}\\n'.format(num+1, all_diseases[all_diseases.diseaseId == i].diseaseName.values[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Intellectual Disability\n",
      "1 Malignant neoplasm of breast\n",
      "2 Schizophrenia\n",
      "3 Epileptic encephalopathy\n",
      "4 Congenital Disorders of Glycosylation\n",
      "5 Ataxias, Hereditary\n",
      "6 Bipolar Disorder\n",
      "7 Dyskeratosis Congenita\n",
      "8 Malignant neoplasm of prostate\n",
      "9 Colorectal Cancer\n"
     ]
    }
   ],
   "source": [
    "top10_metabolic = open('top10_metabolic.txt', 'w')\n",
    "\n",
    "top10_metabolic_ids = d_metabolic_df.sort_values('a',axis=1,ascending=False).columns[:10]\n",
    "for num, i in enumerate(top10_metabolic_ids):\n",
    "    print(num, all_diseases[all_diseases.diseaseId == i].diseaseName.values[0])\n",
    "    top10_metabolic.write('{} {}\\n'.format(num+1, all_diseases[all_diseases.diseaseId == i].diseaseName.values[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Intellectual Disability\n",
      "1 Epileptic encephalopathy\n",
      "2 Schizophrenia\n",
      "3 Malignant neoplasm of prostate\n",
      "4 Malignant neoplasm of breast\n",
      "5 Prostatic Neoplasms\n",
      "6 Colorectal Cancer\n",
      "7 Microcephaly\n",
      "8 Autistic Disorder\n",
      "9 Ataxias, Hereditary\n"
     ]
    }
   ],
   "source": [
    "top10_pediatric = open('top10_pediatric.txt', 'w')\n",
    "\n",
    "top10_pediatric_ids = d_pediatric_df.sort_values('a',axis=1,ascending=False).columns[:10]\n",
    "for num, i in enumerate(top10_pediatric_ids):\n",
    "    print(num, all_diseases[all_diseases.diseaseId == i].diseaseName.values[0])\n",
    "    top10_pediatric.write('{} {}\\n'.format(num+1, all_diseases[all_diseases.diseaseId == i].diseaseName.values[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Intellectual Disability\n",
      "1 Malignant neoplasm of prostate\n",
      "2 Prostatic Neoplasms\n",
      "3 Schizophrenia\n",
      "4 Charcot-Marie-Tooth Disease\n",
      "5 Epileptic encephalopathy\n",
      "6 Colorectal Cancer\n",
      "7 Malignant neoplasm of breast\n",
      "8 Hyperammonemia\n",
      "9 Cholestasis\n"
     ]
    }
   ],
   "source": [
    "top10_carrier = open('top10_carrier.txt', 'w')\n",
    "\n",
    "top10_carrier_ids = d_carrier_df.sort_values('a',axis=1,ascending=False).columns[:10]\n",
    "for num, i in enumerate(top10_carrier_ids):\n",
    "    print(num, all_diseases[all_diseases.diseaseId == i].diseaseName.values[0])\n",
    "    top10_carrier.write('{} {}\\n'.format(num+1, all_diseases[all_diseases.diseaseId == i].diseaseName.values[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 Intellectual Disability\n",
      "1 Charcot-Marie-Tooth Disease\n",
      "2 Schizophrenia\n",
      "3 Malignant neoplasm of prostate\n",
      "4 Prostatic Neoplasms\n",
      "5 Depressive disorder\n",
      "6 Arthrogryposis\n",
      "7 Bipolar Disorder\n",
      "8 Malignant neoplasm of breast\n",
      "9 Mental Depression\n"
     ]
    }
   ],
   "source": [
    "top10_cardioneuro = open('top10_cardioneuro.txt', 'w')\n",
    "\n",
    "top10_cardioneuro_ids = d_cardioneuro_df.sort_values('a',axis=1,ascending=False).columns[:10]\n",
    "for num, i in enumerate(top10_cardioneuro_ids):\n",
    "    print(num, all_diseases[all_diseases.diseaseId == i].diseaseName.values[0])\n",
    "    top10_cardioneuro.write('{} {}\\n'.format(num+1, all_diseases[all_diseases.diseaseId == i].diseaseName.values[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
