{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotating the variants of genes of the patient from the gencode database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_annotations_filename = \"gencode.v18.annotation.gtf\"\n",
    "variant_filename = 'Metabolic_variants.vcf'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Extracted GTF attributes: ['gene_id', 'transcript_id', 'gene_type', 'gene_status', 'gene_name', 'transcript_type', 'transcript_status', 'transcript_name', 'level', 'havana_gene', 'tag', 'havana_transcript', 'exon_number', 'exon_id', 'ont', 'ccdsid']\n"
     ]
    }
   ],
   "source": [
    "from gtfparse import read_gtf\n",
    "# returns GTF with essential columns such as \"feature\", \"seqname\", \"start\", \"end\"\n",
    "# alongside the names of any optional keys which appeared in the attribute column\n",
    "df = read_gtf(gene_annotations_filename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>source</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>frame</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>...</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>level</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>tag</th>\n",
       "      <th>havana_transcript</th>\n",
       "      <th>exon_number</th>\n",
       "      <th>exon_id</th>\n",
       "      <th>ont</th>\n",
       "      <th>ccdsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14412</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>2</td>\n",
       "      <td>OTTHUMG00000000961.2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>14363</td>\n",
       "      <td>29806</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>2</td>\n",
       "      <td>OTTHUMG00000000958.1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>...</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>2</td>\n",
       "      <td>OTTHUMG00000000959.2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>2</td>\n",
       "      <td>OTTHUMG00000000960.1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>52473</td>\n",
       "      <td>54936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>2</td>\n",
       "      <td>OTTHUMG00000185779.1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    seqname  source feature  start    end  score strand frame  \\\n",
       "0      chr1  HAVANA    gene  11869  14412    NaN      +     0   \n",
       "21     chr1  HAVANA    gene  14363  29806    NaN      -     0   \n",
       "82     chr1  HAVANA    gene  29554  31109    NaN      +     0   \n",
       "92     chr1  HAVANA    gene  34554  36081    NaN      -     0   \n",
       "100    chr1  HAVANA    gene  52473  54936    NaN      +     0   \n",
       "\n",
       "               gene_id      transcript_id  ...   transcript_status  \\\n",
       "0    ENSG00000223972.4  ENSG00000223972.4  ...               KNOWN   \n",
       "21   ENSG00000227232.4  ENSG00000227232.4  ...               KNOWN   \n",
       "82   ENSG00000243485.2  ENSG00000243485.2  ...               NOVEL   \n",
       "92   ENSG00000237613.2  ENSG00000237613.2  ...               KNOWN   \n",
       "100  ENSG00000268020.2  ENSG00000268020.2  ...               KNOWN   \n",
       "\n",
       "    transcript_name level           havana_gene tag havana_transcript  \\\n",
       "0           DDX11L1     2  OTTHUMG00000000961.2                         \n",
       "21           WASH7P     2  OTTHUMG00000000958.1                         \n",
       "82       MIR1302-11     2  OTTHUMG00000000959.2                         \n",
       "92          FAM138A     2  OTTHUMG00000000960.1                         \n",
       "100          OR4G4P     2  OTTHUMG00000185779.1                         \n",
       "\n",
       "    exon_number exon_id ont ccdsid  \n",
       "0                                   \n",
       "21                                  \n",
       "82                                  \n",
       "92                                  \n",
       "100                                 \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filter DataFrame to gene entries on chrY\n",
    "df_genes = df[df[\"feature\"] == \"gene\"]\n",
    "df_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 57445 entries, 0 to 2614559\n",
      "Data columns (total 24 columns):\n",
      "seqname              57445 non-null object\n",
      "source               57445 non-null object\n",
      "feature              57445 non-null object\n",
      "start                57445 non-null int64\n",
      "end                  57445 non-null int64\n",
      "score                0 non-null float32\n",
      "strand               57445 non-null object\n",
      "frame                57445 non-null object\n",
      "gene_id              57445 non-null object\n",
      "transcript_id        57445 non-null object\n",
      "gene_type            57445 non-null object\n",
      "gene_status          57445 non-null object\n",
      "gene_name            57445 non-null object\n",
      "transcript_type      57445 non-null object\n",
      "transcript_status    57445 non-null object\n",
      "transcript_name      57445 non-null object\n",
      "level                57445 non-null object\n",
      "havana_gene          57445 non-null object\n",
      "tag                  57445 non-null object\n",
      "havana_transcript    57445 non-null object\n",
      "exon_number          57445 non-null object\n",
      "exon_id              57445 non-null object\n",
      "ont                  57445 non-null object\n",
      "ccdsid               57445 non-null object\n",
      "dtypes: float32(1), int64(2), object(21)\n",
      "memory usage: 10.7+ MB\n"
     ]
    }
   ],
   "source": [
    "df_genes.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seqname', 'source', 'feature', 'start', 'end', 'score', 'strand',\n",
       "       'frame', 'gene_id', 'transcript_id', 'gene_type', 'gene_status',\n",
       "       'gene_name', 'transcript_type', 'transcript_status', 'transcript_name',\n",
       "       'level', 'havana_gene', 'tag', 'havana_transcript', 'exon_number',\n",
       "       'exon_id', 'ont', 'ccdsid'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_genes.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['chr1', 'chr2', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8',\n",
       "       'chr9', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15',\n",
       "       'chr16', 'chr17', 'chr18', 'chr19', 'chr20', 'chr21', 'chr22',\n",
       "       'chrX', 'chrY', 'chrM'], dtype=object)"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_genes['seqname'].unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Look into the list of variants and find correspondance to the gene from the gencode file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "import vcf\n",
    "vcf_reader = vcf.Reader(open(variant_filename, 'r'))\n",
    "\n",
    "import pandas as pd\n",
    "variants = pd.DataFrame(columns=['CHROM', 'start', 'end', 'strand', 'gene_id', 'transcript_id', 'gene_type', 'gene_status',\n",
    "       'gene_name', 'transcript_type', 'transcript_status', 'transcript_name', 'POS','REF','ALT','FILTER'])\n",
    "\n",
    "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   CHROM     start       end strand             gene_id       transcript_id  \\\n",
      "95     1  19197926  19229275      -  ENSG00000159423.12  ENSG00000159423.12   \n",
      "96     1  19175767  19247615      -   ENSG00000255275.3   ENSG00000255275.3   \n",
      "97     1  19197926  19229275      -  ENSG00000159423.12  ENSG00000159423.12   \n",
      "98     1  19175767  19247615      -   ENSG00000255275.3   ENSG00000255275.3   \n",
      "99     1  19197926  19229275      -  ENSG00000159423.12  ENSG00000159423.12   \n",
      "\n",
      "         gene_type gene_status      gene_name transcript_type  \\\n",
      "95  protein_coding       KNOWN        ALDH4A1  protein_coding   \n",
      "96  protein_coding       NOVEL  RP13-279N23.2  protein_coding   \n",
      "97  protein_coding       KNOWN        ALDH4A1  protein_coding   \n",
      "98  protein_coding       NOVEL  RP13-279N23.2  protein_coding   \n",
      "99  protein_coding       KNOWN        ALDH4A1  protein_coding   \n",
      "\n",
      "   transcript_status transcript_name       POS REF  ALT FILTER  \n",
      "95             KNOWN         ALDH4A1  19202770   T  [C]     []  \n",
      "96             NOVEL   RP13-279N23.2  19202896   G  [A]     []  \n",
      "97             KNOWN         ALDH4A1  19202896   G  [A]     []  \n",
      "98             NOVEL   RP13-279N23.2  19202917   T  [C]     []  \n",
      "99             KNOWN         ALDH4A1  19202917   T  [C]     []  \n",
      "    CHROM     start       end strand             gene_id       transcript_id  \\\n",
      "195     1  40538379  40563375      -  ENSG00000131238.11  ENSG00000131238.11   \n",
      "196     1  41445007  41478235      +   ENSG00000171793.9   ENSG00000171793.9   \n",
      "197     1  41445007  41478235      +   ENSG00000171793.9   ENSG00000171793.9   \n",
      "198     1  41445007  41478235      +   ENSG00000171793.9   ENSG00000171793.9   \n",
      "199     1  41445007  41478235      +   ENSG00000171793.9   ENSG00000171793.9   \n",
      "\n",
      "          gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "195  protein_coding       KNOWN      PPT1  protein_coding             KNOWN   \n",
      "196  protein_coding       KNOWN     CTPS1  protein_coding             KNOWN   \n",
      "197  protein_coding       KNOWN     CTPS1  protein_coding             KNOWN   \n",
      "198  protein_coding       KNOWN     CTPS1  protein_coding             KNOWN   \n",
      "199  protein_coding       KNOWN     CTPS1  protein_coding             KNOWN   \n",
      "\n",
      "    transcript_name       POS REF   ALT FILTER  \n",
      "195            PPT1  40562993   T   [C]     []  \n",
      "196           CTPS1  41449291  CT   [C]     []  \n",
      "197           CTPS1  41450911   T   [C]     []  \n",
      "198           CTPS1  41453286  GT   [G]     []  \n",
      "199           CTPS1  41453950   C  [CT]     []  \n",
      "    CHROM     start       end strand            gene_id      transcript_id  \\\n",
      "295     1  85731931  85742773      -  ENSG00000142867.8  ENSG00000142867.8   \n",
      "296     1  85731931  85742773      -  ENSG00000142867.8  ENSG00000142867.8   \n",
      "297     1  92940319  92952433      -  ENSG00000162676.7  ENSG00000162676.7   \n",
      "298     1  92940319  92952433      -  ENSG00000162676.7  ENSG00000162676.7   \n",
      "299     1  92940319  92952433      -  ENSG00000162676.7  ENSG00000162676.7   \n",
      "\n",
      "          gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "295  protein_coding       KNOWN     BCL10  protein_coding             KNOWN   \n",
      "296  protein_coding       KNOWN     BCL10  protein_coding             KNOWN   \n",
      "297  protein_coding       KNOWN      GFI1  protein_coding             KNOWN   \n",
      "298  protein_coding       KNOWN      GFI1  protein_coding             KNOWN   \n",
      "299  protein_coding       KNOWN      GFI1  protein_coding             KNOWN   \n",
      "\n",
      "    transcript_name       POS REF             ALT FILTER  \n",
      "295           BCL10  85741921   G             [C]     []  \n",
      "296           BCL10  85742338   G             [C]     []  \n",
      "297            GFI1  92941357   A             [G]     []  \n",
      "298            GFI1  92944188   G             [A]     []  \n",
      "299            GFI1  92944314   C  [CAG, CAGAGAG]     []  \n",
      "    CHROM      start        end strand            gene_id      transcript_id  \\\n",
      "395     1  120202421  120286838      +  ENSG00000092621.7  ENSG00000092621.7   \n",
      "396     1  120202421  120286838      +  ENSG00000092621.7  ENSG00000092621.7   \n",
      "397     1  120290619  120311528      -  ENSG00000134240.7  ENSG00000134240.7   \n",
      "398     1  120290619  120311528      -  ENSG00000134240.7  ENSG00000134240.7   \n",
      "399     1  120290619  120311528      -  ENSG00000134240.7  ENSG00000134240.7   \n",
      "\n",
      "          gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "395  protein_coding       KNOWN     PHGDH  protein_coding             KNOWN   \n",
      "396  protein_coding       KNOWN     PHGDH  protein_coding             KNOWN   \n",
      "397  protein_coding       KNOWN    HMGCS2  protein_coding             KNOWN   \n",
      "398  protein_coding       KNOWN    HMGCS2  protein_coding             KNOWN   \n",
      "399  protein_coding       KNOWN    HMGCS2  protein_coding             KNOWN   \n",
      "\n",
      "    transcript_name        POS REF  ALT FILTER  \n",
      "395           PHGDH  120284669   A  [T]     []  \n",
      "396           PHGDH  120285546   G  [A]     []  \n",
      "397          HMGCS2  120293352   T  [C]     []  \n",
      "398          HMGCS2  120293367   T  [C]     []  \n",
      "399          HMGCS2  120293568   G  [A]     []  \n",
      "    CHROM      start        end strand             gene_id  \\\n",
      "495     1  160259063  160313190      -  ENSG00000122218.10   \n",
      "496     1  160259063  160313190      -  ENSG00000122218.10   \n",
      "497     1  160259063  160313190      -  ENSG00000122218.10   \n",
      "498     1  160259063  160313190      -  ENSG00000122218.10   \n",
      "499     1  160259063  160313190      -  ENSG00000122218.10   \n",
      "\n",
      "          transcript_id       gene_type gene_status gene_name transcript_type  \\\n",
      "495  ENSG00000122218.10  protein_coding       KNOWN      COPA  protein_coding   \n",
      "496  ENSG00000122218.10  protein_coding       KNOWN      COPA  protein_coding   \n",
      "497  ENSG00000122218.10  protein_coding       KNOWN      COPA  protein_coding   \n",
      "498  ENSG00000122218.10  protein_coding       KNOWN      COPA  protein_coding   \n",
      "499  ENSG00000122218.10  protein_coding       KNOWN      COPA  protein_coding   \n",
      "\n",
      "    transcript_status transcript_name        POS    REF  ALT FILTER  \n",
      "495             KNOWN            COPA  160302444      G  [A]     []  \n",
      "496             KNOWN            COPA  160303288      C  [T]     []  \n",
      "497             KNOWN            COPA  160303416      G  [A]     []  \n",
      "498             KNOWN            COPA  160303542      C  [A]     []  \n",
      "499             KNOWN            COPA  160303755  ATGTG  [A]     []  \n",
      "    CHROM      start        end strand             gene_id  \\\n",
      "595     1  203181955  203242769      -  ENSG00000133063.11   \n",
      "596     1  206940947  206945839      -   ENSG00000136634.5   \n",
      "597     1  206940947  206945839      -   ENSG00000136634.5   \n",
      "598     1  206940947  206945839      -   ENSG00000136634.5   \n",
      "599     1  206940947  206945839      -   ENSG00000136634.5   \n",
      "\n",
      "          transcript_id       gene_type gene_status gene_name transcript_type  \\\n",
      "595  ENSG00000133063.11  protein_coding       KNOWN     CHIT1  protein_coding   \n",
      "596   ENSG00000136634.5  protein_coding       KNOWN      IL10  protein_coding   \n",
      "597   ENSG00000136634.5  protein_coding       KNOWN      IL10  protein_coding   \n",
      "598   ENSG00000136634.5  protein_coding       KNOWN      IL10  protein_coding   \n",
      "599   ENSG00000136634.5  protein_coding       KNOWN      IL10  protein_coding   \n",
      "\n",
      "    transcript_status transcript_name        POS REF  ALT FILTER  \n",
      "595             KNOWN           CHIT1  203194186   C  [T]     []  \n",
      "596             KNOWN            IL10  206943351   T  [C]     []  \n",
      "597             KNOWN            IL10  206944233   A  [G]     []  \n",
      "598             KNOWN            IL10  206944645   T  [C]     []  \n",
      "599             KNOWN            IL10  206944861   A  [C]     []  \n",
      "    CHROM      start        end strand            gene_id      transcript_id  \\\n",
      "695     1  235824341  236046940      -  ENSG00000143669.9  ENSG00000143669.9   \n",
      "696     1  235824341  236046940      -  ENSG00000143669.9  ENSG00000143669.9   \n",
      "697     1  235824341  236046940      -  ENSG00000143669.9  ENSG00000143669.9   \n",
      "698     1  235824341  236046940      -  ENSG00000143669.9  ENSG00000143669.9   \n",
      "699     1  235824341  236046940      -  ENSG00000143669.9  ENSG00000143669.9   \n",
      "\n",
      "          gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "695  protein_coding       KNOWN      LYST  protein_coding             KNOWN   \n",
      "696  protein_coding       KNOWN      LYST  protein_coding             KNOWN   \n",
      "697  protein_coding       KNOWN      LYST  protein_coding             KNOWN   \n",
      "698  protein_coding       KNOWN      LYST  protein_coding             KNOWN   \n",
      "699  protein_coding       KNOWN      LYST  protein_coding             KNOWN   \n",
      "\n",
      "    transcript_name        POS     REF               ALT FILTER  \n",
      "695            LYST  235906983       T               [C]     []  \n",
      "696            LYST  235909485       T               [C]     []  \n",
      "697            LYST  235916140       C               [T]     []  \n",
      "698            LYST  235918688  CAAAAG  [C, CAAAAGAAAAG]     []  \n",
      "699            LYST  235922228       T               [C]     []  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    CHROM    start      end strand            gene_id      transcript_id  \\\n",
      "795     2  9628615  9695921      -  ENSG00000151694.8  ENSG00000151694.8   \n",
      "796     2  9628615  9695921      -  ENSG00000151694.8  ENSG00000151694.8   \n",
      "797     2  9628615  9695921      -  ENSG00000151694.8  ENSG00000151694.8   \n",
      "798     2  9628615  9695921      -  ENSG00000151694.8  ENSG00000151694.8   \n",
      "799     2  9628615  9695921      -  ENSG00000151694.8  ENSG00000151694.8   \n",
      "\n",
      "          gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "795  protein_coding       KNOWN    ADAM17  protein_coding             KNOWN   \n",
      "796  protein_coding       KNOWN    ADAM17  protein_coding             KNOWN   \n",
      "797  protein_coding       KNOWN    ADAM17  protein_coding             KNOWN   \n",
      "798  protein_coding       KNOWN    ADAM17  protein_coding             KNOWN   \n",
      "799  protein_coding       KNOWN    ADAM17  protein_coding             KNOWN   \n",
      "\n",
      "    transcript_name      POS REF  ALT FILTER  \n",
      "795          ADAM17  9661058   C  [A]     []  \n",
      "796          ADAM17  9661067   A  [T]     []  \n",
      "797          ADAM17  9661072   A  [G]     []  \n",
      "798          ADAM17  9661074  CA  [C]     []  \n",
      "799          ADAM17  9661080   T  [C]     []  \n",
      "    CHROM     start       end strand            gene_id      transcript_id  \\\n",
      "895     2  31557187  31637581      -  ENSG00000158125.5  ENSG00000158125.5   \n",
      "896     2  31557187  31637581      -  ENSG00000158125.5  ENSG00000158125.5   \n",
      "897     2  31557187  31637581      -  ENSG00000158125.5  ENSG00000158125.5   \n",
      "898     2  31557187  31637581      -  ENSG00000158125.5  ENSG00000158125.5   \n",
      "899     2  31557187  31637581      -  ENSG00000158125.5  ENSG00000158125.5   \n",
      "\n",
      "          gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "895  protein_coding       KNOWN       XDH  protein_coding             KNOWN   \n",
      "896  protein_coding       KNOWN       XDH  protein_coding             KNOWN   \n",
      "897  protein_coding       KNOWN       XDH  protein_coding             KNOWN   \n",
      "898  protein_coding       KNOWN       XDH  protein_coding             KNOWN   \n",
      "899  protein_coding       KNOWN       XDH  protein_coding             KNOWN   \n",
      "\n",
      "    transcript_name       POS    REF  ALT FILTER  \n",
      "895             XDH  31590917      T  [C]     []  \n",
      "896             XDH  31591384  ACAGG  [A]     []  \n",
      "897             XDH  31591761      T  [C]     []  \n",
      "898             XDH  31592895      C  [T]     []  \n",
      "899             XDH  31592955      T  [C]     []  \n",
      "    CHROM     start       end strand             gene_id       transcript_id  \\\n",
      "995     2  44543420  44589001      -  ENSG00000138078.11  ENSG00000138078.11   \n",
      "996     2  44543420  44589001      -  ENSG00000138078.11  ENSG00000138078.11   \n",
      "997     2  44543420  44589001      -  ENSG00000138078.11  ENSG00000138078.11   \n",
      "998     2  44543420  44589001      -  ENSG00000138078.11  ENSG00000138078.11   \n",
      "999     2  47143296  47303276      +  ENSG00000068724.11  ENSG00000068724.11   \n",
      "\n",
      "          gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "995  protein_coding       KNOWN     PREPL  protein_coding             KNOWN   \n",
      "996  protein_coding       KNOWN     PREPL  protein_coding             KNOWN   \n",
      "997  protein_coding       KNOWN     PREPL  protein_coding             KNOWN   \n",
      "998  protein_coding       KNOWN     PREPL  protein_coding             KNOWN   \n",
      "999  protein_coding       KNOWN     TTC7A  protein_coding             KNOWN   \n",
      "\n",
      "    transcript_name       POS   REF  ALT FILTER  \n",
      "995           PREPL  44571215  CAAT  [C]     []  \n",
      "996           PREPL  44573624     G  [A]     []  \n",
      "997           PREPL  44573658     T  [C]     []  \n",
      "998           PREPL  44573671     G  [T]     []  \n",
      "999           TTC7A  47177787     G  [T]     []  \n",
      "     CHROM      start        end strand            gene_id      transcript_id  \\\n",
      "1095     2   98330023   98356325      +  ENSG00000115085.9  ENSG00000115085.9   \n",
      "1096     2   98330023   98356325      +  ENSG00000115085.9  ENSG00000115085.9   \n",
      "1097     2   98330023   98356325      +  ENSG00000115085.9  ENSG00000115085.9   \n",
      "1098     2   98330023   98356325      +  ENSG00000115085.9  ENSG00000115085.9   \n",
      "1099     2  109510927  109605828      -  ENSG00000135960.5  ENSG00000135960.5   \n",
      "\n",
      "           gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "1095  protein_coding       KNOWN     ZAP70  protein_coding             KNOWN   \n",
      "1096  protein_coding       KNOWN     ZAP70  protein_coding             KNOWN   \n",
      "1097  protein_coding       KNOWN     ZAP70  protein_coding             KNOWN   \n",
      "1098  protein_coding       KNOWN     ZAP70  protein_coding             KNOWN   \n",
      "1099  protein_coding       KNOWN      EDAR  protein_coding             KNOWN   \n",
      "\n",
      "     transcript_name        POS REF  ALT FILTER  \n",
      "1095           ZAP70   98341195   C  [T]     []  \n",
      "1096           ZAP70   98341728   C  [G]     []  \n",
      "1097           ZAP70   98351986   T  [C]     []  \n",
      "1098           ZAP70   98356250   G  [C]     []  \n",
      "1099            EDAR  109513321   C  [A]     []  \n",
      "     CHROM      start        end strand            gene_id      transcript_id  \\\n",
      "1195     2  171669723  171717661      +  ENSG00000128683.9  ENSG00000128683.9   \n",
      "1196     2  171669723  171717661      +  ENSG00000128683.9  ENSG00000128683.9   \n",
      "1197     2  172290727  172341562      +  ENSG00000115827.9  ENSG00000115827.9   \n",
      "1198     2  172290727  172341562      +  ENSG00000115827.9  ENSG00000115827.9   \n",
      "1199     2  172290727  172341562      +  ENSG00000115827.9  ENSG00000115827.9   \n",
      "\n",
      "           gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "1195  protein_coding       KNOWN      GAD1  protein_coding             KNOWN   \n",
      "1196  protein_coding       KNOWN      GAD1  protein_coding             KNOWN   \n",
      "1197  protein_coding       KNOWN    DCAF17  protein_coding             KNOWN   \n",
      "1198  protein_coding       KNOWN    DCAF17  protein_coding             KNOWN   \n",
      "1199  protein_coding       KNOWN    DCAF17  protein_coding             KNOWN   \n",
      "\n",
      "     transcript_name        POS REF     ALT FILTER  \n",
      "1195            GAD1  171678952   C     [T]     []  \n",
      "1196            GAD1  171700917  CT     [C]     []  \n",
      "1197          DCAF17  172291616   C     [T]     []  \n",
      "1198          DCAF17  172299859   T    [TA]     []  \n",
      "1199          DCAF17  172304939   T  [TAAC]     []  \n",
      "     CHROM      start        end strand             gene_id  \\\n",
      "1295     2  191829084  191885686      -  ENSG00000115415.14   \n",
      "1296     2  191829084  191885686      -  ENSG00000115415.14   \n",
      "1297     2  191829084  191885686      -  ENSG00000115415.14   \n",
      "1298     2  191829084  191885686      -  ENSG00000115415.14   \n",
      "1299     2  191829084  191885686      -  ENSG00000115415.14   \n",
      "\n",
      "           transcript_id       gene_type gene_status gene_name  \\\n",
      "1295  ENSG00000115415.14  protein_coding       KNOWN     STAT1   \n",
      "1296  ENSG00000115415.14  protein_coding       KNOWN     STAT1   \n",
      "1297  ENSG00000115415.14  protein_coding       KNOWN     STAT1   \n",
      "1298  ENSG00000115415.14  protein_coding       KNOWN     STAT1   \n",
      "1299  ENSG00000115415.14  protein_coding       KNOWN     STAT1   \n",
      "\n",
      "     transcript_type transcript_status transcript_name        POS     REF  \\\n",
      "1295  protein_coding             KNOWN           STAT1  191850099       A   \n",
      "1296  protein_coding             KNOWN           STAT1  191850566       T   \n",
      "1297  protein_coding             KNOWN           STAT1  191851997       T   \n",
      "1298  protein_coding             KNOWN           STAT1  191859653   AATAT   \n",
      "1299  protein_coding             KNOWN           STAT1  191859654  ATATAT   \n",
      "\n",
      "      ALT FILTER  \n",
      "1295  [G]     []  \n",
      "1296  [C]     []  \n",
      "1297  [C]     []  \n",
      "1298  [A]     []  \n",
      "1299  [A]     []  \n",
      "     CHROM      start        end strand            gene_id      transcript_id  \\\n",
      "1395     2  219940051  220035549      -  ENSG00000213901.6  ENSG00000213901.6   \n",
      "1396     2  219940039  220025587      -  ENSG00000187736.8  ENSG00000187736.8   \n",
      "1397     2  219940051  220035549      -  ENSG00000213901.6  ENSG00000213901.6   \n",
      "1398     2  219940039  220025587      -  ENSG00000187736.8  ENSG00000187736.8   \n",
      "1399     2  219940051  220035549      -  ENSG00000213901.6  ENSG00000213901.6   \n",
      "\n",
      "           gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "1395  protein_coding       KNOWN   SLC23A3  protein_coding             KNOWN   \n",
      "1396  protein_coding       KNOWN     NHEJ1  protein_coding             KNOWN   \n",
      "1397  protein_coding       KNOWN   SLC23A3  protein_coding             KNOWN   \n",
      "1398  protein_coding       KNOWN     NHEJ1  protein_coding             KNOWN   \n",
      "1399  protein_coding       KNOWN   SLC23A3  protein_coding             KNOWN   \n",
      "\n",
      "     transcript_name        POS REF  ALT FILTER  \n",
      "1395         SLC23A3  220022074   A  [G]     []  \n",
      "1396           NHEJ1  220022176   C  [T]     []  \n",
      "1397         SLC23A3  220022176   C  [T]     []  \n",
      "1398           NHEJ1  220022435   G  [C]     []  \n",
      "1399         SLC23A3  220022435   G  [C]     []  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     CHROM      start        end strand            gene_id      transcript_id  \\\n",
      "1495     2  241807896  241819919      +  ENSG00000172482.4  ENSG00000172482.4   \n",
      "1496     2  241807896  241819919      +  ENSG00000172482.4  ENSG00000172482.4   \n",
      "1497     2  241807896  241819919      +  ENSG00000172482.4  ENSG00000172482.4   \n",
      "1498     2  241807896  241819919      +  ENSG00000172482.4  ENSG00000172482.4   \n",
      "1499     2  242295658  242434256      +  ENSG00000006607.8  ENSG00000006607.8   \n",
      "\n",
      "           gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "1495  protein_coding       KNOWN      AGXT  protein_coding             KNOWN   \n",
      "1496  protein_coding       KNOWN      AGXT  protein_coding             KNOWN   \n",
      "1497  protein_coding       KNOWN      AGXT  protein_coding             KNOWN   \n",
      "1498  protein_coding       KNOWN      AGXT  protein_coding             KNOWN   \n",
      "1499  protein_coding       KNOWN     FARP2  protein_coding             KNOWN   \n",
      "\n",
      "     transcript_name        POS REF  ALT FILTER  \n",
      "1495            AGXT  241817272   G  [T]     []  \n",
      "1496            AGXT  241817516   A  [G]     []  \n",
      "1497            AGXT  241817927   C  [T]     []  \n",
      "1498            AGXT  241818527   A  [C]     []  \n",
      "1499           FARP2  242393373  AT  [A]     []  \n",
      "     CHROM     start       end strand            gene_id      transcript_id  \\\n",
      "1595     3  31574130  31679112      +  ENSG00000163527.5  ENSG00000163527.5   \n",
      "1596     3  31574130  31679112      +  ENSG00000163527.5  ENSG00000163527.5   \n",
      "1597     3  31574130  31679112      +  ENSG00000163527.5  ENSG00000163527.5   \n",
      "1598     3  31574130  31679112      +  ENSG00000163527.5  ENSG00000163527.5   \n",
      "1599     3  31574130  31679112      +  ENSG00000163527.5  ENSG00000163527.5   \n",
      "\n",
      "           gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "1595  protein_coding       KNOWN     STT3B  protein_coding             KNOWN   \n",
      "1596  protein_coding       KNOWN     STT3B  protein_coding             KNOWN   \n",
      "1597  protein_coding       KNOWN     STT3B  protein_coding             KNOWN   \n",
      "1598  protein_coding       KNOWN     STT3B  protein_coding             KNOWN   \n",
      "1599  protein_coding       KNOWN     STT3B  protein_coding             KNOWN   \n",
      "\n",
      "     transcript_name       POS REF           ALT FILTER  \n",
      "1595           STT3B  31659779   T  [TTACCTAACA]     []  \n",
      "1596           STT3B  31663515   A      [AGTGTG]     []  \n",
      "1597           STT3B  31666162  CT           [C]     []  \n",
      "1598           STT3B  31666256  CT           [C]     []  \n",
      "1599           STT3B  31667291   A           [G]     []  \n",
      "     CHROM     start       end strand            gene_id      transcript_id  \\\n",
      "1695     3  69024365  69063112      -  ENSG00000163378.9  ENSG00000163378.9   \n",
      "1696     3  69024365  69063112      -  ENSG00000163378.9  ENSG00000163378.9   \n",
      "1697     3  69024365  69063112      -  ENSG00000163378.9  ENSG00000163378.9   \n",
      "1698     3  69024365  69063112      -  ENSG00000163378.9  ENSG00000163378.9   \n",
      "1699     3  69024365  69063112      -  ENSG00000163378.9  ENSG00000163378.9   \n",
      "\n",
      "           gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "1695  protein_coding       KNOWN      EOGT  protein_coding             KNOWN   \n",
      "1696  protein_coding       KNOWN      EOGT  protein_coding             KNOWN   \n",
      "1697  protein_coding       KNOWN      EOGT  protein_coding             KNOWN   \n",
      "1698  protein_coding       KNOWN      EOGT  protein_coding             KNOWN   \n",
      "1699  protein_coding       KNOWN      EOGT  protein_coding             KNOWN   \n",
      "\n",
      "     transcript_name       POS REF          ALT FILTER  \n",
      "1695            EOGT  69054611   A          [G]     []  \n",
      "1696            EOGT  69056604   G          [A]     []  \n",
      "1697            EOGT  69056658   C         [CG]     []  \n",
      "1698            EOGT  69056662   G  [GGGGGGGGC]     []  \n",
      "1699            EOGT  69058783   C          [T]     []  \n",
      "     CHROM      start        end strand            gene_id      transcript_id  \\\n",
      "1795     3  135969148  136056738      +  ENSG00000114054.9  ENSG00000114054.9   \n",
      "1796     3  135969148  136056738      +  ENSG00000114054.9  ENSG00000114054.9   \n",
      "1797     3  135969148  136056738      +  ENSG00000114054.9  ENSG00000114054.9   \n",
      "1798     3  135969148  136056738      +  ENSG00000114054.9  ENSG00000114054.9   \n",
      "1799     3  135969148  136056738      +  ENSG00000114054.9  ENSG00000114054.9   \n",
      "\n",
      "           gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "1795  protein_coding       KNOWN      PCCB  protein_coding             KNOWN   \n",
      "1796  protein_coding       KNOWN      PCCB  protein_coding             KNOWN   \n",
      "1797  protein_coding       KNOWN      PCCB  protein_coding             KNOWN   \n",
      "1798  protein_coding       KNOWN      PCCB  protein_coding             KNOWN   \n",
      "1799  protein_coding       KNOWN      PCCB  protein_coding             KNOWN   \n",
      "\n",
      "     transcript_name        POS REF          ALT FILTER  \n",
      "1795            PCCB  136003158   T  [TGATGTTTC]     []  \n",
      "1796            PCCB  136003159   T          [A]     []  \n",
      "1797            PCCB  136003194   A          [C]     []  \n",
      "1798            PCCB  136016478   A          [G]     []  \n",
      "1799            PCCB  136035737   A          [G]     []  \n",
      "     CHROM     start       end strand            gene_id      transcript_id  \\\n",
      "1895     4  39460620  39479273      +  ENSG00000121897.9  ENSG00000121897.9   \n",
      "1896     4  39460620  39479273      +  ENSG00000121897.9  ENSG00000121897.9   \n",
      "1897     4  39460620  39479273      +  ENSG00000121897.9  ENSG00000121897.9   \n",
      "1898     4  39460620  39479273      +  ENSG00000121897.9  ENSG00000121897.9   \n",
      "1899     4  39460620  39479273      +  ENSG00000121897.9  ENSG00000121897.9   \n",
      "\n",
      "           gene_type gene_status gene_name transcript_type transcript_status  \\\n",
      "1895  protein_coding       KNOWN      LIAS  protein_coding             KNOWN   \n",
      "1896  protein_coding       KNOWN      LIAS  protein_coding             KNOWN   \n",
      "1897  protein_coding       KNOWN      LIAS  protein_coding             KNOWN   \n",
      "1898  protein_coding       KNOWN      LIAS  protein_coding             KNOWN   \n",
      "1899  protein_coding       KNOWN      LIAS  protein_coding             KNOWN   \n",
      "\n",
      "     transcript_name       POS REF        ALT FILTER  \n",
      "1895            LIAS  39462088   A        [G]     []  \n",
      "1896            LIAS  39462153   C  [CA, CAA]     []  \n",
      "1897            LIAS  39463472   G        [A]     []  \n",
      "1898            LIAS  39464816   T        [C]     []  \n",
      "1899            LIAS  39466639   A        [G]     []  \n",
      "     CHROM      start        end strand             gene_id  \\\n",
      "1995     4  108910870  108956331      +  ENSG00000138796.11   \n",
      "1996     4  108910870  108956331      +  ENSG00000138796.11   \n",
      "1997     4  108910870  108956331      +  ENSG00000138796.11   \n",
      "1998     4  108910870  108956331      +  ENSG00000138796.11   \n",
      "1999     4  108910870  108956331      +  ENSG00000138796.11   \n",
      "\n",
      "           transcript_id       gene_type gene_status gene_name  \\\n",
      "1995  ENSG00000138796.11  protein_coding       KNOWN      HADH   \n",
      "1996  ENSG00000138796.11  protein_coding       KNOWN      HADH   \n",
      "1997  ENSG00000138796.11  protein_coding       KNOWN      HADH   \n",
      "1998  ENSG00000138796.11  protein_coding       KNOWN      HADH   \n",
      "1999  ENSG00000138796.11  protein_coding       KNOWN      HADH   \n",
      "\n",
      "     transcript_type transcript_status transcript_name        POS REF   ALT  \\\n",
      "1995  protein_coding             KNOWN            HADH  108935769   T  [TC]   \n",
      "1996  protein_coding             KNOWN            HADH  108935770   T   [C]   \n",
      "1997  protein_coding             KNOWN            HADH  108935771   T   [C]   \n",
      "1998  protein_coding             KNOWN            HADH  108935871   G   [A]   \n",
      "1999  protein_coding             KNOWN            HADH  108935970   C   [G]   \n",
      "\n",
      "          FILTER  \n",
      "1995          []  \n",
      "1996          []  \n",
      "1997  [QdFilter]  \n",
      "1998          []  \n",
      "1999          []  \n",
      "     CHROM      start        end strand             gene_id  \\\n",
      "2095     4  186990306  187006255      +   ENSG00000164342.7   \n",
      "2096     4  186990306  187006255      +   ENSG00000164342.7   \n",
      "2097     5    1253262    1295184      -  ENSG00000164362.14   \n",
      "2098     5    1253262    1295184      -  ENSG00000164362.14   \n",
      "2099     5    1253262    1295184      -  ENSG00000164362.14   \n",
      "\n",
      "           transcript_id       gene_type gene_status gene_name  \\\n",
      "2095   ENSG00000164342.7  protein_coding       KNOWN      TLR3   \n",
      "2096   ENSG00000164342.7  protein_coding       KNOWN      TLR3   \n",
      "2097  ENSG00000164362.14  protein_coding       KNOWN      TERT   \n",
      "2098  ENSG00000164362.14  protein_coding       KNOWN      TERT   \n",
      "2099  ENSG00000164362.14  protein_coding       KNOWN      TERT   \n",
      "\n",
      "     transcript_type transcript_status transcript_name        POS REF    ALT  \\\n",
      "2095  protein_coding             KNOWN            TLR3  187005379   C   [CT]   \n",
      "2096  protein_coding             KNOWN            TLR3  187005683   G   [GA]   \n",
      "2097  protein_coding             KNOWN            TERT    1271524   T    [C]   \n",
      "2098  protein_coding             KNOWN            TERT    1271661   T    [C]   \n",
      "2099  protein_coding             KNOWN            TERT    1272688   C  [CCA]   \n",
      "\n",
      "          FILTER  \n",
      "2095  [QdFilter]  \n",
      "2096          []  \n",
      "2097          []  \n",
      "2098          []  \n",
      "2099          []  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     CHROM     start       end strand             gene_id       transcript_id  \\\n",
      "2195     5  41870132  41871059      +   ENSG00000248668.1   ENSG00000248668.1   \n",
      "2196     5  52391509  52405893      -  ENSG00000164172.14  ENSG00000164172.14   \n",
      "2197     5  52391509  52405893      -  ENSG00000164172.14  ENSG00000164172.14   \n",
      "2198     5  52405672  52410956      +   ENSG00000247796.2   ENSG00000247796.2   \n",
      "2199     5  58264865  59817947      -  ENSG00000113448.12  ENSG00000113448.12   \n",
      "\n",
      "                 gene_type gene_status      gene_name       transcript_type  \\\n",
      "2195  processed_transcript       NOVEL      OXCT1-AS1  processed_transcript   \n",
      "2196        protein_coding       KNOWN          MOCS2        protein_coding   \n",
      "2197        protein_coding       KNOWN          MOCS2        protein_coding   \n",
      "2198             antisense       NOVEL  CTD-2366F13.1             antisense   \n",
      "2199        protein_coding       KNOWN          PDE4D        protein_coding   \n",
      "\n",
      "     transcript_status transcript_name       POS REF  ALT FILTER  \n",
      "2195             NOVEL       OXCT1-AS1  41870675   T  [C]     []  \n",
      "2196             KNOWN           MOCS2  52397586   C  [T]     []  \n",
      "2197             KNOWN           MOCS2  52405693   C  [A]     []  \n",
      "2198             NOVEL   CTD-2366F13.1  52405693   C  [A]     []  \n",
      "2199             KNOWN           PDE4D  59759657   C  [T]     []  \n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m----------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                    Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-40-b381c1816d22>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mrow\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdf_genes\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitertuples\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m         \u001b[0;31m#print(str(chrom), row.seqname[3:])\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m         \u001b[0mthere_is_entry\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mchrom\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseqname\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mpos\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstart\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mpos\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mend\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mthere_is_entry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m             \u001b[0mnew_row\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m'POS'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mpos\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'start'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstart\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'end'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mend\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'strand'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstrand\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'gene_id'\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgene_id\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'transcript_id'\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtranscript_id\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'gene_type'\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgene_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'gene_status'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgene_status\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'gene_name'\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgene_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'transcript_type'\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtranscript_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'transcript_status'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtranscript_status\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'transcript_name'\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtranscript_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'CHROM'\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mchrom\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'POS'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mpos\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'REF'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mref\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'ALT'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0malt\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'FILTER'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mfilter_pass\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "counter = 0\n",
    "for record in vcf_reader:\n",
    "    chrom = record.CHROM\n",
    "    pos = record.POS\n",
    "    ref = record.REF\n",
    "    alt = record.ALT\n",
    "    filter_pass = record.FILTER\n",
    "    for row in df_genes.itertuples():\n",
    "        #print(str(chrom), row.seqname[3:])\n",
    "        there_is_entry = str(chrom) == row.seqname[3:] and pos > row.start and pos < row.end\n",
    "        if there_is_entry:\n",
    "            new_row = {'POS': pos, 'start': row.start, 'end': row.end, 'strand': row.strand, 'gene_id' : row.gene_id, 'transcript_id' : row.transcript_id, 'gene_type' : row.gene_type, 'gene_status': row.gene_status, 'gene_name' : row.gene_name, 'transcript_type' : row.transcript_type, 'transcript_status': row.transcript_status, 'transcript_name' : row.transcript_name, 'CHROM' : chrom, 'POS': pos,'REF':ref,'ALT':alt,'FILTER':filter_pass}\n",
    "            variants = variants.append(new_row, ignore_index=True)\n",
    "            counter += 1\n",
    "            #print(counter)\n",
    "            if counter % 100 == 0:\n",
    "                print(variants.tail())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "        loc = row.geo_location\n",
    "        mask = distance(df_o['geo_location'].values, loc) < 100\n",
    "        objects_nearby = df_o.loc[mask]\n",
    "    # do something with this neighborhood"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
